### PR TITLE
[mono] Enable arm64 intrinsic tests

### PR DESCRIPTION
**Pull request by imhameed**
This change:

- Removes `base` from the list of LLVM-style attributes supported by the AOT
compiler.

- Adds another icall that raises a `PlatformNotSupportedException`, and bumps
the AOT file format version number.

- Makes unsupported intrinsics raise a `PlatformNotSupportedException`
regardless of the codegen backend used: AOT/JIT/interpreter, mini/LLVM.

- Enables formerly-disabled tests in `JIT/HardwareIntrinsics/Arm/**/*`.

- Incrementally consolidates duplicated arm64/amd64 intrinsics-handling code.
Intrinsic "group" lookup is now table-driven, and most hardware intrinsic
lookup logic happens in one place.

### CODE REVIEW

**Comment by imhameed** [comment]
The arm64 failures are relevant, although I haven't (yet) reproduced the exact same failure on an actual arm64 Linux machine

File: src/tests/Common/CLRTest.Execute.Bash.targets [hunk 1]
```diff
@@ -305,6 +305,36 @@ if [ ! -z ${RunCrossGen+x} ]%3B then
...
+    LAUNCHER="$_DebuggerFullPath $(_CLRTestRunFile)"
+fi
+
+$(BashIlrtTestLaunchCmds)
+
+if [ ! -z ${RunCrossGen+x} ]%3B then
+  TakeLock
+fi
+
+export MONO_ENV_OPTIONS=--llvm
```
Comment (lambdageek) [inline src/tests/Common/CLRTest.Execute.Bash.targets:327]: unintentional?

**Review by lambdageek — COMMENTED** [review]
LGTM, but I'm not an intrinsics expert

File: src/tests/Common/CLRTest.Execute.Bash.targets (hunk 1 above)
Comment (fanyang-mono) [reply src/tests/Common/CLRTest.Execute.Bash.targets:327]: This is done intentionally, so that for LLVM AOT arm64 lane, it falls back to LLVM JIT.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -318,6 +320,99 @@ type_to_insert_op (MonoType *type)
 	}
 }
 
+struct IntrinGroup {
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:323]: We usually add typedefs for structure types to avoid having to write 'struct' everywhere.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 3]
```diff
@@ -318,6 +320,99 @@ type_to_insert_op (MonoType *type)
...
+    const char *name;
+    MonoCPUFeatures feature;
+    const SimdIntrinsic *intrinsics;
+    int intrinsics_size;
+    gboolean jit_supported;
+};
+
+struct EmitIntrinsicCtx {
+	gboolean is_64bit;
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:332]: Why is this needed ? The arguments have to be packed/unpacked around the call and its only used for 1 function, the emit function.

**Review by vargaz — APPROVED** [review]

File: src/tests/Common/CLRTest.Execute.Bash.targets (hunk 1 above)
Comment (lambdageek) [reply src/tests/Common/CLRTest.Execute.Bash.targets:327]: why would an AOT lane use the JIT?

Also if a CI lane needs a particular configuration, it should set some properties and then this file should set MONO_ENV_OPTIONS based on the value of the property.

Comment (lambdageek) [reply src/tests/Common/CLRTest.Execute.Bash.targets:327]: Oh wait, this is new stuff, not modifications of an existing configuration.
Nevermind.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 3 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c]: I am guessing the reason why this is needed is because otherwise we will have to pass all those arguments into `emit_hardware_intrinsics` separately. Then, the function signature will be super long.
